### DISCUSSION TITLE
Dark mode image source switch?

### DISCUSSION THREAD

**Discussion by jrappen** [vuepress/core#1084] [Q&A] [answered]
I tried:

```vue
<img
    :src="document.documentElement.classList.contains('dark') ? $withBase('/images/hero.dark.svg') : $withBase('/images/hero.svg')"
    alt="Logo"
    loading="lazy"
>
```

on a markdown page and GitHub CI (with `v2.0.0-beta.51`) chokes with a:

```text
TypeError: Cannot read properties of undefined (reading documentElement)
```

Is there a quick fix to this?

**Comment by Mister-Hope** [comment, marked as answer]
Use md-enhance plugin. It has a feature called image mark.

https://vuepress-theme-hope.github.io/v2/md-enhance/guide/image.html#image-mark
